optuna/storages/_in_memory.py

```diff
@@ -471,3 +472,106 @@ def __init__(self, name: str) -> None:
         self.name: str = name
         self.directions: List[StudyDirection] = [StudyDirection.NOT_SET]
         self.best_trial_id: Optional[int] = None
+
+
+class _Manager(BaseManager):
+    pass
+
+
+_Manager.register("InMemoryStorage", InMemoryStorage)
+
+
+class MultiprocessInMemoryStorage(BaseStorage):
+    def __init__(self) -> None:
+        self._manager = _Manager()
+        self._manager.start()
```

Comment:
[QUESTION] Don't we need to explicitly call the `BaseManager.shutdown()` method to stop the server when the storage instance is dropped?

Reply:
That's a good point. I thought so too and actually implemented a finalizer in an earlier commit, but later reverted it https://github.com/optuna/optuna/pull/2759/commits/b59af0bb0e33f14ee39537ef21f945f40e12ebab since it was already handled by the library (but it's allowed to call `shutdown` multiple times).

> Manager processes will be shutdown as soon as they are garbage collected or their parent process exits. The manager classes are defined in the multiprocessing.managers module:

https://docs.python.org/3.6/library/multiprocessing.html

I think it's enough, but do you think we need to implement our own logic?

Reply:
Got it! Thanks for your explanation.
I also think that the current implementation is enough, but how about leaving a short comment about the behavior for future readers like me?

Reply:
Sounds good, I added a comment here https://github.com/optuna/optuna/pull/2759/commits/47012a4d754f8d7c26821d85d8fcfc4239da1762.

Reply:
👍